lib/ansible/runner.rb

```diff
@@ -317,11 +319,11 @@ def fetch_galaxy_roles(playbook_or_role_args)
         return unless playbook_or_role_args[:playbook]
 
         playbook_dir = File.dirname(playbook_or_role_args[:playbook])
-        Ansible::Content.new(playbook_dir).fetch_galaxy_roles
+        Ansible::Content.new(playbook_dir).fetch_galaxy_roles(runner_env)
       end
 
       def runner_env
-        {"PYTHONPATH" => python_path}.delete_nils
+        {"PYTHONPATH" => runner_python_path, "PATH" => runner_path}.compact
```

Comment:
Technically, we should have been setting the path before as well, which is why this change works on the old python3.9 and the new python3.12.  On either version, the virtual env activate sets both of these.